### PR TITLE
update rpm provider with new options

### PR DESCRIPTION
**Pull request by cristifalcas**

### CODE REVIEW

**Comment by alexjfisher** [comment]
@cristifalcas If you rebase, the puppet-lint travis errors should go away

**Comment by alexjfisher** [comment]
@cristifalcas 
I've just tested creating a repo with `download_policy => on_demand` and `repoview => true`.  Worked great!

**Comment by alexjfisher** [comment]
Actually, the `repoview` parameter doesn't seem to be idempotent.
Every time I run puppet...
```
Notice: /Stage[main]/Main/Pulp_rpmrepo[epel-7]/repoview: repoview changed 'true' to 'true'
```

**Comment by ekohl** [comment]
The code looks good to me. @alexjfisher can you confirm the latest commit fixes it for you?

**Comment by alexjfisher** [comment]
@ekohl Tested it, and the change looks good.  Thanks @cristifalcas!

File: lib/puppet/provider/pulp_rpmrepo/api.rb [hunk 1]
```diff
@@ -75,35 +75,42 @@ def self.get_resource_properties(repo_id)
...
-        hash[:proxy_pass]       = importer['config']['proxy_password']
-        hash[:max_downloads]    = importer['config']['max_downloads']
-        hash[:max_speed]        = importer['config']['max_speed']
-        hash[:remove_missing]   = importer['config']['remove_missing'] ? :true : :false
-        hash[:retain_old_count] = importer['config']['retain_old_count']
+        hash[:basicauth_pass]    = importer['config']['basic_auth_password'] || ''
+        hash[:basicauth_user]    = importer['config']['basic_auth_username'] || ''
+        hash[:download_policy]   = importer['config']['download_policy']
+        hash[:allowed_keys]      = importer['config']['allowed_keys'] || ''
+        hash[:require_signature] = importer['config']['require_signature'] || ''
```
Comment (ekohl) [inline lib/puppet/provider/pulp_rpmrepo/api.rb]: In the types this is a boolean. Should this also have ```? :true : :false```?

Comment (alexjfisher) [inline lib/puppet/provider/pulp_rpmrepo/api.rb]: Looks that way.  That parameter is also not idempotent right now.
```
Notice: /Pulp_rpmrepo[centos-6-x86_64-extras]/require_signature: require_signature changed 'true' to 'true'
```

**Comment by ekohl** [comment]
@cristifalcas could you fix the require_signature parameter? Other than that I think it can be merged.

**Comment by alexjfisher** [comment]
ping @cristifalcas

**Comment by ekohl** [comment]
Merged with the require_signature true/false fix. Thanks @cristifalcas and @alexjfisher!

**Pull request closed without merging**